docs/pages/modules/overview.mdx

```diff
@@ -13,61 +13,38 @@ Expo provides a set of APIs and utilities to improve the process of developing n
+To make the maintainance easier and let the community iterate faster, we decided to create a module system that reduces the costs and technical debt to a minimum.
 
-## Use the Expo Modules API in an existing React Native library
+### Why Swift and Kotlin?
 
-You may want to use the Expo Modules API in existing React Native libraries, for example with [AppDelegate Subscribers](./appdelegate-subscribers.mdx) you can hook into `AppDelegate` methods without requiring developers to copy any code over to their own `AppDelegate`. This is particularly useful to add seamless support for Expo managed projects to a library. The following steps will set up your existing React Native library to have access to the Expo Modules API.
+After several years of maintaining over 50 native modules in the Expo SDK, we have found out that many issues that have arisen were caused by unhandled null values or using incorrect types.
+Modern language features can help developers avoid these bugs; for example, the lack of optional types combined with the dynamism of Objective-C made it tough to catch certain classes of bugs that would have been caught by the compiler in Swift.
 
-### 1. Initialize the module config
+Another difficulty of writing React Native modules is context switching between the vastly different languages and paradigms for writing native modules on each platform.
+Due to the differences between these platforms, it cannot be completely avoided, but we feel the need to have just one common API and documentation to simplify the development as much as possible and make it easier for a single developer to maintain a library on multiple platforms.
 
-Create the [module config](module-config) **expo-module.config.json** file just near your **package.json** and start from the empty object `{}` in there. We will fill it in later to enable specific features. <br/>
+This is one of the reasons why the Expo Modules ecosystem was designed from the ground up to be used with modern native languages: Swift and Kotlin.
 
-### 2. Add the `expo-modules-core` native dependency
+### Passing data between runtimes
 
-Add `expo-modules-core` as a dependency in your podspec and **build.gradle** files.<br/>
+Another big pain point that we have encountered is the validation of arguments passed from JavaScript to native functions.
+This is especially error prone, time consuming, and difficult to maintain when it comes to `NSDictionary` or `ReadableMap`, where the type of values is unknown in runtime and each property needs to be validated separately by the developer.
+A valuable feature of the Expo native modules API is that it has full knowledge of the argument types the native function expects, so it can pre-validate and convert the arguments for you! The dictionaries can be represented as native structs that we call [Records](./module-api.md#records).
```

Comment:
```suggestion
A valuable feature of the Expo native modules API is its full knowledge of the argument types the native function expe. It can pre-validate and convert the arguments for you! The dictionaries can be represented as native structs that we call [Records](./module-api.md#records).
```